server/src/main/java/org/elasticsearch/repositories/RepositoriesStatsCollector.java

```diff
@@ -0,0 +1,155 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.repositories;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.cluster.service.ClusterService;
+import org.elasticsearch.common.component.AbstractLifecycleComponent;
+import org.elasticsearch.common.settings.Setting;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.common.util.concurrent.AbstractRunnable;
+import org.elasticsearch.threadpool.Scheduler;
+import org.elasticsearch.threadpool.ThreadPool;
+
+import java.util.Map;
+import java.util.function.Consumer;
+import java.util.function.Supplier;
+
+public final class RepositoriesStatsCollector extends AbstractLifecycleComponent {
+
+    private static final Logger logger = LogManager.getLogger(RepositoriesStatsCollector.class);
+
+    public static final TimeValue MIN_INTERVAL = TimeValue.timeValueSeconds(1L);
+
+    public static final Setting<TimeValue> INTERVAL =
+        Setting.timeSetting("repositories.monitoring.export.interval", TimeValue.timeValueMinutes(1L), MIN_INTERVAL,
+            Setting.Property.Dynamic, Setting.Property.NodeScope);
+
+    public static final Setting<Boolean> ENABLED =
+        Setting.boolSetting("repositories.monitoring.export.enabled", true,
+            Setting.Property.Dynamic, Setting.Property.NodeScope);
+
+    private final StatsCollector statsCollector = new StatsCollector();
+
+    private final Supplier<Map<String, RepositoryStats>> statsProvider;
+    private final ThreadPool threadPool;
+    private final Consumer<Map<String, RepositoryStats>> exporter;
+    private final ClusterService clusterService;
+    private volatile TimeValue interval;
+    private volatile boolean enabled;
+    private volatile Scheduler.Cancellable scheduler;
+
+    public RepositoriesStatsCollector(final Settings settings,
+                                      final ClusterService clusterService,
```

Comment:
No need for the dependency on the `ClusterService` here I think, just the `ClusterSettings` should be good enough right?